snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/event/AbstractEvent.kt

```diff
@@ -38,15 +55,15 @@ abstract class AbstractEvent : Event {
     
     // Builder methods
 
-    /** Adds a list of context entities to the existing ones. */
-    fun entities(entities: List<SelfDescribingJson>?): AbstractEvent {
-        entities?.let { customContexts.addAll(entities) }
+    /** Replace the context entities attached to the event with a new list of entities. */
+    fun entities(entities: List<SelfDescribingJson>): AbstractEvent {
+        this.entities = entities.toMutableList()
```

Comment:
I am a bit of a coward and have a bad feeling about this 😅 On one hand, this looks like a builder function which usually replace the values, but going from a behaviour that is non-destructive (appending) to a destructive one (replacing) is a nasty breaking change. Nobody will notice this in the migration guide and it may lead to some very hard to find bugs with missing data.

I think I would prefer if we kept the Android behaviour and updated on iOS to append instead of replace. Other changes in this PR look good, I'd keep those. What do you think?

Reply:
That's fair! I do think the API makes more sense as full replacement. But being cautious is good.